core/templates/pages/topics-and-skills-dashboard-page/topics-list/topics-list.directive.ts

```diff
@@ -47,19 +50,24 @@ angular.module('oppia').directive('topicsList', [
             $scope, $uibModal, $rootScope, EditableTopicBackendApiService,
             AlertsService, EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED) {
           var ctrl = this;
-          $scope.getTopicEditorUrl = function(topicId) {
+          ctrl.getTopicEditorUrl = function(topicId) {
             return '/topic_editor/' + topicId;
           };
+          ctrl.showEditOptions = function(topicId) {
+            return ctrl.selectedIndex === topicId;
+          };
+
+          ctrl.enableEditOptions = function(topicId) {
+            ctrl.selectedIndex = topicId;
+          };
 
-          $scope.selectTopic = function(topicId) {
-            if ($scope.selectedTopicIds) {
-              if ($scope.selectedTopicIds.indexOf(topicId) === -1) {
-                $scope.selectedTopicIds.push(topicId);
-              }
-            }
+          ctrl.getSerialNumber = function(index) {
```

Comment:
"serial number" is not descriptive, and neither is "index".  Please choose clearer names if possible and also write JSDoc for each function you add.

Reply:
Done.